### PR TITLE
Swift 2 support

### PR DESCRIPTION
**Pull request by AndrewSB**
Add support for swift 2

This shouldn't be on `master`, but on a `swift-2` branch. Unfortunately, I can not create a pull request to a branch that doesn't exist yet. @hackiftekhar If you create a swift-2 branch, I'll edit this PR to merge into that instead

### CODE REVIEW

**Comment by hackiftekhar** [comment]
Actually I'm planning to first update swift code to match Obj-C version. Then will update to swift-2. Let me know what you think about it.

**Comment by AndrewSB** [comment]
That sounds totally fine, do you think a swift-2 branch can be maintained?
If you'd like I can keep it up to date. Otherwise I'll have to keep my fork.
Furthermore I think other people will also benefit from this swift-2 version

Let me know what you think

**Comment by hackiftekhar** [comment]
I can also think that we need to maintain swift-2 branch till apple officially release swift2.0 and Xcode7

**Comment by AndrewSB** [comment]
I'm sorry @hackiftekhar, I don't understand what you said. Can you rephrase please?

**Comment by hackiftekhar** [comment]
Sure!

I simply mean that Xcode7 is in beta phase, and we can't provide official support for Swift2.0 till Apple release Xcode7 for general use. Till then we can create swift2 branch and can maintain swift2 version there like you suggested. Let me know if you still have any confusion.

**Comment by AndrewSB** [comment]
yes, that sounds perfect. If you create a `swift2` branch from `master` right now I can edit this PR/ create a new one that targets it.

**Comment by hackiftekhar** [comment]
Swift2 branch just created for you. :)

**Comment by hackiftekhar** [comment]
FYI, I've updated swift code to match current ObjC version

**Comment by AndrewSB** [comment]
Thank you for you work @hackiftekhar! I'll create a pull to the `swift2` branch
